imap_processing/hi/l1b/hi_l1b.py

```diff
@@ -341,3 +341,32 @@ def compute_hae_coordinates(dataset: xr.Dataset) -> dict[str, xr.DataArray]:
     new_vars["hae_longitude"].values = pointing_coordinates[:, 1]
 
     return new_vars
+
+
+def de_esa_energy_step(dataset: xr.Dataset) -> dict[str, xr.DataArray]:
```

Comment:
Just clarification question. Is this function to figure out ESA voltages of input energy steps?

Reply:
Not quite. The `esa_step` in telemetry references an index in the sweep table that was used. The sweep table determines the voltages that the ESA inner and outer are set to and thus the energy bandpass of what particles get through. I don't quite understand all of the details of the algorithm that will be implemented in this function yet. But it will use the housekeeping data to look at voltages and then maybe map those voltages to an energy step. This ticket has some additional detail: #779